README.md

```diff
@@ -1,2 +1,85 @@
 # business-analytics
 Tools to produce Riff Business analytic reports on RiffEdu and RiffPlatform deployments
+
+This repository uses a third-party package called [Metabase][] to visualize usage metrics for Mattermost. Two docker containers are run; one using the metabase image and another using the
+official mysql image. Commands are provided to restore sql database backups from Mattermost to the mysql container.
+
+## Getting Started
+
+These instructions will get you a copy of the project up and running on your local machine for development and testing purposes.
+
+### Database Backups
+
+Download sql backups from the Riff Learning backup location (ask for the URL if you don't have it), and place all .sql files in the folder called 'database-backups' in the root of this repo.
+
+### Prerequisites
+
+- [docker CE][docker-install] (Community Edition)
+- [docker-compose][docker-compose-install]
+- git
+- make
+
+
+## Running the Software
+The first step is to clone this repository. The following commands are defined in a Makefile, and can be run as follows:
+
+```
+make up
+```
+This command will create the mysql and metabase containers and show the logs for each
+
+```
+make up-detach
+```
+This command will create the mysql and metabase containers in detached mode (no logs).
+
+```
+make down
+```
+This command will stop and remove the mysql and metabase containers. The mysql volume containing the sql backups will
```

Comment:
My initial impression here was that you were suggesting make clean would run something like 'docker volume prune', after you run make down. Perhaps this command would also be useful to have in the Makefile? 

I have added 'make clean' to the Makefile which deletes all .sql files in the folder ./database-backups. I have added to the README a recommendation to run this command after restoring the backups to the mysql container.

Reply:
The `docker volume prune` might also be nice to have as a make target, but the deletion one is what I was hoping for.

Reply:
`docker volume prune` would be very bad to put in a target in the Makefile!
You would likely end up deleting volumes for containers completely unrelated to this project.
You have named volumes. Remove it/them explicitly.
`docker volume rm ba-mmdb-data`

A `make clean` which deletes the files which are manually obtained isn't great. When this is integrated w/ John's scripts used perhaps in a `make update-db` target, which downloads the latest backup automatically, then `make clean` should be removing that/those and probably the mysql volume it is restored into.

Reply:
Ah, agreed explicitly removing the relevant volumes makes more sense.

Reply:
I am not 100% clear on your second paragraph. What are you recommending I implement right now? My interpretation is: `make clean` should be run when you are done using the software, and it should delete the mysql backups as well as the mysql volume?

Reply:
`clean` is a [common Makefile target](https://www.gnu.org/prep/standards/html_node/Standard-Targets.html) with a common somewhat expected meaning
> Delete all files in the current directory that are normally created by building the program. Also delete files in other directories if they are created by this makefile.
Basically it cleans up things created by the makefile. The `.sql` files are not created by the makefile and therefore the `clean` target shouldn't remove them. It would be fine if a different target removed them.

Is that clearer?

Reply:
Ok makes sense. You were suggesting, I think, that in the future, when a script downloads the sql backups, then` make clean` should remove them, since at that point, they are created/downloaded dynamically.

But for now, maybe `make clean` should:

1. Remove the metebase log file (metabase.db.trace.db)
2. run `docker volume rm ba-mmdb-data`

Reply:
yep that's exactly what I was saying.

I personally don't think I'd bother w/ that additional target, but if you and John think it's useful, that is the way to do it.

Reply:
Ok makes sense. Thanks!

Reply:
It looks like the default behaviour when using docker-compose, is that the directory name is prepended to the volume name. So in this case, the volume is officially called `business-analytics_ba-mmdb-data`, and thus the command to delete will be `docker volume rm business-analytics_ba-mmdb-data`. Is it safe to say that the directory will always be `business-analytics`? Technically somebody could clone the repo into a folder with a different name.

Reply:
What I'd suggest is that either you get the directory name yourself, or that you assume `business-analytics` for now and leave a comment that is what you're doing.

Reply:
Ok makes sense. I've addressed this in a new commit.